### ISSUE TITLE
Duplicate entries in oplog.timestamp file

### ISSUE THREAD

**Issue by XDestination** [mongodb-labs/mongo-connector#316]
I am using the mongo-connector to replicate some data from a 7-shard cluster to a single instance of mongoDB. Today I realized that the number of documents between the two databases is drifting apart and therefore I started to investigate.

The first thing that looks weird to me is that there are duplicate entries in the oplog.timestamp file with different values for the timestamp:

```
[
	["Collection(Database(MongoClient([u'server01.abc.com:27018', u'server02.abc.com:27018', u'server03.abc.com:27018', u'server04.abc.com:27018']), u'local'), u'oplog.rs')", 6187759245908246585], 
	["Collection(Database(MongoClient([u'server05.abc.com:27018', u'server06.abc.com:27018', u'server07.abc.com:27018', u'server08.abc.com:27018']), u'local'), u'oplog.rs')", 6189300503512350739],
	["Collection(Database(MongoClient([u'server05.abc.com:27018', u'server06.abc.com:27018', u'server08.abc.com:27018', u'server07.abc.com:27018']), u'local'), u'oplog.rs')", 6190048532196491279], 
	["Collection(Database(MongoClient([u'server09.abc.com:27018', u'server10.abc.com:27018', u'server11.abc.com:27018', u'server12.abc.com:27018']), u'local'), u'oplog.rs')", 6190048532196491282], 
	["Collection(Database(MongoClient([u'server13.abc.com:27018', u'server14.abc.com:27018', u'server15.abc.com:27018']), u'local'), u'oplog.rs')", 6190048115584663561],
	["Collection(Database(MongoClient([u'server16.abc.com:27018', u'server17.abc.com:27018', u'server18.abc.com:27018', u'server19.abc.com:27018']), u'local'), u'oplog.rs')", 6190048523606556710],
	["Collection(Database(MongoClient([u'server16.abc.com:27018', u'server19.abc.com:27018', u'server18.abc.com:27018', u'server17.abc.com:27018']), u'local'), u'oplog.rs')", 6189298321668964353],
	["Collection(Database(MongoClient([u'server20.abc.com:27018', u'server21.abc.com:27018', u'server22.abc.com:27018', u'server23.abc.com:27018']), u'local'), u'oplog.rs')", 6190048515016622104],
	["Collection(Database(MongoClient([u'server24.abc.com:27018', u'server25.abc.com:27018', u'server26.abc.com:27018', u'server27.abc.com:27018']), u'local'), u'oplog.rs')", 6187716682782343169]
]
```
Any thoughts on how this can happen? Is this causing problems? How can I prevent this?

**Comment by XDestination** [comment]
I assume it's happening, when the nodes in a replica set change.
Would it be possible to change the oplog-progress file to contain a list of (replica-set-name, last-oplog-timestamp) pairs?
The duplicate entries are causing problems on my side, because I have checks to monitor the replication lag and those duplicate entries always produce false positives (since they are not being updated).

**Comment by aherlihy** [comment]
#399 Will fix this but needs to be backwards compatible.

**Comment by llvtt** [comment]
This should be resolved by #469, can you confirm @XDestination? Thanks again for your hard work on helping getting this resolved.

**Comment by llvtt** [comment]
I'm assuming this has been fixed, since there have been no recent issues to the contrary. Please feel free to open a new bug if this issue rears its head again.
